libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/speech/MapboxSpeechPlayer.java

```diff
@@ -0,0 +1,233 @@
+package com.mapbox.services.android.navigation.ui.v5.voice.speech;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.os.AsyncTask;
+import android.text.TextUtils;
+
+import com.amazonaws.auth.CognitoCachingCredentialsProvider;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.services.android.navigation.ui.v5.voice.InstructionListener;
+import com.mapbox.services.android.navigation.ui.v5.voice.InstructionPlayer;
+import com.mapbox.services.android.navigation.v5.navigation.VoiceInstructionLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import okhttp3.Cache;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import timber.log.Timber;
+
+/**
+ * <p>
+ * Will retrieve synthesized speech mp3s from Mapbox's API Voice.
+ * </p>
+ */
+public class MapboxSpeechPlayer implements InstructionPlayer {
+  private static final long TEN_MB_CACHE_SIZE = 10 * 1098 * 1098;
+  private static final String SSML_TEXT_TYPE = "ssml";
+  private static final String ERROR_TEXT = "Unable to set data source for the media mediaPlayer! %s";
+  private Queue<File> instructionQueue;
+  private VoiceInstructionLoader voiceInstructionLoader;
+  private MediaPlayer mediaPlayer;
+  private InstructionListener instructionListener;
+  private boolean isMuted;
+  private String cacheDirectory;
+
+  /**
+   * Construct an instance of {@link MapboxSpeechPlayer}
+   *
+   * @param context   to initialize {@link CognitoCachingCredentialsProvider} and {@link AudioManager}
+   */
+  public MapboxSpeechPlayer(Context context, Locale locale) {
+    this.cacheDirectory = context.getCacheDir().toString();
+    instructionQueue = new ConcurrentLinkedQueue();
+    voiceInstructionLoader = VoiceInstructionLoader.builder()
+      .language(locale.toString())
+      .cache(new Cache(context.getCacheDir(), TEN_MB_CACHE_SIZE))
+      .accessToken(Mapbox.getAccessToken())
+      .build();
+  }
+
+  /**
+   * Sets the listener to listen for instruction events
+   *
+   * @param instructionListener listener to set
+   */
+  public void setInstructionListener(InstructionListener instructionListener) {
+    this.instructionListener = instructionListener;
+  }
+
+  /**
+   * Plays the specified text instruction using MapboxSpeech API, defaulting to SSML input type
+   *
+   * @param instruction voice instruction to be synthesized and played
+   */
+  @Override
+  public void play(String instruction) {
+    play(instruction, SSML_TEXT_TYPE);
+  }
+
+  /**
+   * Plays the specified text instruction using MapboxSpeech API
+   *
+   * @param instruction voice instruction to be synthesized and played
+   * @param textType either "ssml" or "text"
+   */
+  public void play(String instruction, String textType) {
```

Comment:
> Access can be private

Why are we exposing this? It seems unnecessary.

Reply:
If anyone was going to use MapboxSpeechPlayer by itself and wanted to use text instructions intstead of SSML they would need this, but not sure if that's an actual use case. I can make it private for now